### PR TITLE
Improving the Installation Guide

### PR DESCRIPTION
**Pull request by israel-hdez**
Epic https://github.com/kiali/kiali/issues/4118
As part of kiali/kiali#3858.

* Break the single-page installation guide into several pages by topic.
* Improve structure of the installation guide.
* Remove content that is not related to installation (it will be moved
  to other parts of the docs).
* Simplify the quick start (it was too complex to be quick).
* Beautify "lightbulbs" and "bullhorns" of touched content with [adoc admonitions](https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#admonitions).

There is some removed content that is not related with the installation. I'll think where to place it and resurrect it in a follow up PR.

Updated content preview:
* Quick start: https://deploy-preview-415--kiali.netlify.app/documentation/staging/quick-start/
* Installation guide: https://deploy-preview-415--kiali.netlify.app/documentation/staging/installation-guide/

### CODE REVIEW

File: content/documentation/staging/installation-guide/accessing-kiali.adoc [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+:sectlinks:
+:linkattrs:
+
+toc::[]
+
+== Introduction
+
+Once Kiali is succesfully installed, you can use your favorite method to access
+or expose Kiali. For your convenience, this page describes a few methods to
+expose or access Kiali.
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/accessing-kiali.adoc:21]:
```suggestion
After Kiali is succesfully installed you will need to make Kiali accessible to users.  This page describes some popular methods of exposing Kiali for use.
```

File: content/documentation/staging/installation-guide/accessing-kiali.adoc [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+If you will be using your own exposure method, you may need to
+<<route-configs,set some configurations>> to make Kiali aware about how users
+are reaching Kiali.
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/accessing-kiali.adoc:25]:
```suggestion
If exposing Kiali in a custom way,  you may need to
<<route-configs,set some configurations>> to make Kiali aware of how users
will access Kiali.
```

File: content/documentation/staging/installation-guide/accessing-kiali.adoc [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+NOTE: The examples on this page assume that you followed the link:{{< ref
+"../installation-guide" >}}[Installation guide] to install Kiali and that you
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/accessing-kiali.adoc:28]:
```suggestion
"../installation-guide" >}}[Installation guide] to install Kiali, and that you
```

File: content/documentation/staging/installation-guide/accessing-kiali.adoc [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+installed Kiali in the `istio-system` namespace.
+
+== Accessing Kiali using port forwarding
+
+TIP: This method should work in any Kubernetes cluster kind.
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/accessing-kiali.adoc:33]:
```suggestion
TIP: This method should work in any kind of Kubernetes cluster.
```

File: content/documentation/staging/installation-guide/accessing-kiali.adoc [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+# If you have kubectl command line tool
+kubectl port-forward svc/kiali 20001:20001 -n istio-system
+----
+
+These commands will block. Access Kiali by visiting `\https://localhost:2001/` in
+your preferred web browser.
+
+CAUTION: Please note that this method is not exposing Kiali to any user. It is
+only making it available in your machine, given that you have enough privileges
+to perform port forwarding.
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/accessing-kiali.adoc:50]:
```suggestion
CAUTION: Please note that this method exposes Kiali *only* to the local machine, no external users.  You must
have the necessary privileges to perform port forwarding.
```

File: content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+Assuming you saved the previous YAML to a file named `my-kiali-cr.yaml`, you
+create the resource in the `istio-system` namespace with the following command:
+
+[source,bash]
+----
+$ kubectl apply -f my-kiali-cr.yaml -n istio-system
+----
+
+TIP: Typically, the Kiali CR is created in the Istio namespace. Thus, also Kiali is installed in this namespace.
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc:40]:
```suggestion
Assuming you saved the previous YAML to a file named `my-kiali-cr.yaml`, and that you are
installing Kiali in the same default namespace as Istio, create the resource with the following command:

[source,bash]
----
$ kubectl apply -f my-kiali-cr.yaml -n istio-system
----

TIP:  Often, but not always, Kiali is installed in the same namespace as Istio, thus the Kiali CR is also created in the Istio namespace.
```

File: content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+Once created, the Kiali Operator will process the resource and perform Kiali
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc:42]:
```suggestion
Once created, the Kiali Operator should shortly be notified and will process the resource,  performing the Kiali
```

File: content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    Kubernetes Version:  1.21.2
+    Operator Version:    v1.40.0
+  Progress:
+    Duration:  0:00:16
+    Message:   5. Creating core resources
+Events:        <none>
+----
+
+IMPORTANT: All resources created by the Kiali Operator are managed by the
+Operator and should *not* be manually edited.
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc:78]:
```suggestion
IMPORTANT: *Never* manually edit resources created by the Kiali Operator, only the Kiali CR.
```

File: content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+We recommended to download the
+link:https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml[example
+Kiali CR YAML file] that is available in the Operator's GitHub repository. This
+example file contains and describes all available settings. Then, edit the
+downloaded file being careful to maintain proper formatting.
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc:84]:
```suggestion
downloaded file being *very careful* to maintain proper formatting.  Incorrect indentation is a common problem!
```

File: content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+NOTE: The link in the previous paragraph is for the example Kiali CR hosted in
+GitHub, in the `master` branch. Use GitHub's branch selector to see the example
+file that matches the Kiali Operator version that you are using.
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc:88]: This is what we need to get away from, and get the example CR into kiali.io.

File: content/documentation/staging/installation-guide/install-with-helm.adoc [hunk 11]
```diff
@@ -0,0 +1,283 @@
+---
+title: "Installation with Helm"
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/install-with-helm.adoc:2]:
```suggestion
title: "Installation via Helm"
```

File: content/documentation/staging/installation-guide/installing-with-operatorhub.adoc [hunk 12]
```diff
@@ -0,0 +1,52 @@
+---
+title: "Installation with the OperatorHub"
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/installing-with-operatorhub.adoc:2]:
```suggestion
title: "Installation via OperatorHub"
```

**Review by jshaughn — CHANGES_REQUESTED** [review]
@israel-hdez Nice reorg, it's in line with the Features change and is a better layout of content.  And makes updating individual sections more natural.  The updated lightbulbs/bullhorns etc is good.   I edited in place and generated a bunch of suggestions for you to consider.  I'm sure several of them are for existing content, not anything you changed, but may still be worthwhile changes.

File: content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc (hunk 10 above)
Comment (israel-hdez) [reply content/documentation/staging/installation-guide/creating-updating-kiali-cr.adoc:88]: Yes, I know... I think we need to keep this until everything is on kiali.io.

File: content/documentation/staging/installation-guide/_index.adoc [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+  kubectl rollout restart deployments kiali -n istio-system
+----
+
+=== Updating Permissions
+
+==== Secrets Permissions
+
+Kiali and its operator have reduced permissions for handling secrets due to security reasons.
+
+For some features like link:/documentation/staging/features/istio-component-status/#_certificates_information_indicators[Certificates Information Indicators], read permissions for specific secrets in the control plane namespace might be required.
```
[the remark this answers is not in the document]
Comment (israel-hdez) [reply content/documentation/staging/installation-guide/_index.adoc]: This is commented out. I'll anyway apply this fix as I plan to move around and resurrect this content.

File: content/documentation/staging/installation-guide/_index.adoc [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+    certificates_information_indicators:
+      enabled: true
+      secrets:
+      - dns.example1-service-account
+      - dns.example2-service-account
+----
+
+These permissions can be configured in two different ways depending on how you install the operator.
+
+When installing the operator using the Helm chart, set the property `secretReader`. Secrets listed in this property will be allowed to be read by the operator and subsequently by Kiali.
```
[the remark this answers is not in the document]
Comment (israel-hdez) [reply content/documentation/staging/installation-guide/_index.adoc]: Same here. This is commented out. I'll anyway apply this fix...

File: content/documentation/staging/installation-guide/_index.adoc [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+
+By default the secrets listed are:
+
+[source,yaml]
+----
+secretReader: ['cacerts', 'istio-ca-secret']
+----
+
+When installing the operator with OLM there is no way to pre-configure the operator permissions prior to installing the operator. The solution is to manually update the cluster role used by the operator as follows:
```
[the remark this answers is not in the document]
Comment (israel-hdez) [reply content/documentation/staging/installation-guide/_index.adoc]: ...and here

File: content/documentation/staging/installation-guide/_index.adoc [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+
+[source,bash]
+----
+kubectl patch $(kubectl get clusterroles -o name | grep kiali-operator) --type "json" -p '[{"op":"add","path":"/rules/0","value":{"apiGroups":[""],"resources":["secrets"],"verbs":["get"],"resourceNames":["secret-name-to-be-read"]}}]'
+----
+
+Where the `secret-name-to-be-read` needs to be replaced by the secret name you want the operator to read.
```
[the remark this answers is not in the document]
Comment (israel-hdez) [reply content/documentation/staging/installation-guide/_index.adoc]: ... and here :-)

**Comment by israel-hdez** [comment]
@jshaughn I merged all your suggestions. I hope everything will be fine now.

**Review by jshaughn — APPROVED** [review]
LGTM

**Comment by israel-hdez** [comment]
I'll merge... I'll think what to do with the PR of the main navigation bar.
Expect a CI failure while I resurrect the content that is currently commented outl.

**Pull request merged**